eddymotion/model.py

```diff
@@ -332,6 +334,59 @@ def predict(self, gradient, **kwargs):
         return retval
 
 
+class SparseFascicleModel:
+    """
+    A wrapper of :obj:`dipy.reconst.sfm.SparseFascicleModel.
+    """
+
+    __slots__ = ("_model", "_S0", "_mask", "_solver")
+
+    def __init__(self, gtab, S0=None, mask=None, solver=None, **kwargs):
+        """Instantiate the wrapped model."""
+        from dipy.reconst.sfm import SparseFascicleModel
+
+        self._S0 = None
+        if S0 is not None:
+            self._S0 = np.clip(
+                S0.astype("float32") / S0.max(),
+                a_min=1e-5,
+                a_max=1.0,
+            )
+
+        self._mask = mask
+        if mask is None and S0 is not None:
+            self._mask = self._S0 > np.percentile(self._S0, 35)
+
+        if self._mask is not None:
+            self._S0 = self._S0[self._mask.astype(bool)]
+
+        self._solver = solver
+        if solver is None:
+            self._solver = "ElasticNet"
+
+        kwargs = {k: v for k, v in kwargs.items() if k in ("solver",)}
+        self._model = SparseFascicleModel(gtab, **kwargs)
+
+    def fit(self, data, **kwargs):
+        """Clean-up permitted args and kwargs, and call model's fit."""
+        self._model = self._model.fit(data[self._mask, ...])
```

Comment:
For parallelization, we can divide the data up into chunks before calling this, and then call this separately on each chunk (on a separate thread/process)

Reply:
Thanks @arokem, I'm going to hand it over to @dpys to work on the chunking in a separate PR

Reply:
@arokem and @oesteban -- Did we ever decide on whether the best approach for this would be to migrate @sebastientourbier 's asyncio parallelization routines for the DTI model to a base class for parallel fit/predict that could be imported to each of the SFM and DKI API's as well? 

On a separate note-- would it be better to stick with asyncio, switch to joblib, or expose both approaches as optional?

Reply:
I would stick with a copy of Sebastien's solution here. Let's worry about a more elegant OO design after we see that this works well here.